src/stream.js

```diff
@@ -1,412 +1,416 @@
-"use strict";
+'use strict';
 
 function Lazy(func) {
-    this.has_evaluated = false;
-    this.func = func;
-    this.value = null;
+  this.has_evaluated = false;
+  this.func = func;
+  this.value = null;
 }
 
 Lazy.prototype = {
-    eval: function() {
-        if ( this.has_evaluated ) {
-            return this.value;
-        }
-
-        this.value = this.func();
-        this.has_evaluated = true;
-        return this.value;
+  eval: function() {
+    if (this.has_evaluated) {
+      return this.value;
     }
+
+    this.value = this.func();
+    this.has_evaluated = true;
+    return this.value;
+  }
 };
 
 function Eager(func) {
-    this.func = func;
+  this.func = func;
 }
 
 Eager.prototype = {
-    eval: function() {
-        return this.func();
-    }
+  eval: function() {
+    return this.func();
+  }
+}
+
+function Stream(head, tailPromise, wrapper) {
+  if (typeof wrapper == 'undefined') {
+    wrapper = Lazy;
+  }
+
+  this.wrapper = wrapper;
+
+  if (typeof head != 'undefined') {
```

Comment:
@dotnetCarpenter I suggest avoiding the creation of another function to check for inequality to avoid unnecessary code implementation. We can simply use the following expression: `if (!isUndefined(value)`.